File: frontend/src/types/splitReceiver.ts
export interface IPageResult<T> {
  records: T[];
  total: number;
  current: number;
  size: number;
}

export interface SplitReceiver {
  id: number;
  receiverNo: string;
  merchantNo: string;
  receiverName: string;
  receiverType: number;
  idCardNo: string;
  idCardName: string;
  bankCardNo: string;
  bankPhone: string;
  bankName: string;
  bankBranchName: string;
  verifyStatus: number;
  verifyChannel: string;
  verifyTime: string;
  verifyFailReason: string;
  verifyRequestId: string;
  contactName: string;
  contactPhone: string;
  contactEmail: string;
  status: number;
  remark: string;
  operatorId: string;
  operatorName: string;
  createdAt: string;
  updatedAt: string;
  verifyStatusDesc: string;
  receiverTypeDesc: string;
  verifyChannelDesc: string;
  statusDesc: string;
}

export interface SplitReceiverSaveRequest {
  receiverNo?: string;
  receiverName: string;
  receiverType: number;
  idCardNo: string;
  idCardName: string;
  bankCardNo: string;
  bankPhone: string;
  bankName: string;
  bankBranchName?: string;
  contactName?: string;
  contactPhone?: string;
  contactEmail?: string;
  remark?: string;
  status?: number;
}

export interface SplitReceiverVerifyRequest {
  receiverNo: string;
  verifyChannel: string;
}

export interface SplitReceiverBatchImportItem {
  receiverName: string;
  receiverType: number;
  idCardNo: string;
  idCardName: string;
  bankCardNo: string;
  bankPhone: string;
  bankName: string;
  bankBranchName?: string;
  contactName?: string;
  contactPhone?: string;
  contactEmail?: string;
  remark?: string;
}

export interface SplitReceiverVerifyLog {
  id: number;
  logNo: string;
  merchantNo: string;
  receiverNo: string;
  verifyChannel: string;
  verifyRequestId: string;
  idCardName: string;
  idCardNo: string;
  bankCardNo: string;
  bankPhone: string;
  verifyStatus: number;
  verifyResult: string;
  verifyFailCode: string;
  verifyFailReason: string;
  verifyTime: string;
  requestData: string;
  responseData: string;
  operatorId: string;
  operatorName: string;
  createdAt: string;
  verifyChannelDesc: string;
  verifyStatusDesc: string;
}

export interface SplitReceiverQueryParams {
  current?: number;
  size?: number;
  receiverNo?: string;
  receiverName?: string;
  receiverType?: number;
  verifyStatus?: number;
  status?: number;
  idCardNo?: string;
  bankCardNo?: string;
}

export interface SplitReceiverVerifyLogQueryParams {
  current?: number;
  size?: number;
  receiverNo?: string;
  verifyChannel?: string;
  verifyStatus?: number;
  verifyRequestId?: string;
}

File: backend/settlement/src/main/java/com/payhub/settlement/entity/SplitReceiver.java
package com.payhub.settlement.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("split_receiver")
public class SplitReceiver implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String receiverNo;

    private String merchantNo;

    private String receiverName;

    private Integer receiverType;

    private String idCardNo;

    private String idCardName;

    private String bankCardNo;

    private String bankPhone;

    private String bankName;

    private String bankBranchName;

    private Integer verifyStatus;

    private Integer verifyChannel;

    private LocalDateTime verifyTime;

    private String verifyFailReason;

    private String verifyFailCode;

    private String verifyRequestId;

    private Integer idCardVerifyChannel;

    private Integer idCardVerifyStatus;

    private LocalDateTime idCardVerifyTime;

    private String idCardVerifyRequestId;

    private String idCardVerifyLevel;

    private String idCardVerifyFailCode;

    private String idCardVerifyFailReason;

    private String contactName;

    private String contactPhone;

    private String contactEmail;

    private Integer status;

    private String remark;

    private String operatorId;

    private String operatorName;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}
